crates/ubrn_cli/src/repo.rs

```diff
@@ -79,21 +79,42 @@ impl GitRepoArgs {
             run_cmd(&mut cmd)?;
         }
 
-        // git fetch --depth 1 origin $branch
+        // git ls-remote origin $branch
+        let output = Command::new("git")
+            .current_dir(self.directory(project_root)?)
+            .arg("ls-remote")
+            .arg("origin")
+            .arg(&self.branch)
+            .output()?;
+        let output = String::from_utf8(output.stdout)?;
+
+        // Find $branch in the output and resolve the SHA or fall back to $branch
+        let mut sha = None;
+        let branch_ref = format!("refs/heads/{}", &self.branch);
+        let tag_ref = format!("refs/tags/{}", &self.branch);
+        for line in output.lines() {
+            if line.ends_with(&branch_ref) || line.ends_with(&tag_ref) {
+                sha = Some(line.split_whitespace().next().unwrap());
+                break;
+            }
+        }
+        let sha = sha.unwrap_or(&self.branch);
```

Comment:
Ooh nice. 

That `let mut sha = None` is a little bit of a smell.

Would you consider:

```suggestion
        // Find $branch in the output and resolve the SHA or fall back to $branch
        let branch_ref = format!("refs/heads/{}", &self.branch);
        let tag_ref = format!("refs/tags/{}", &self.branch);
        let sha = output
            .lines()
            .find(|line| line.ends_with(&branch_ref) || line.ends_with(&tag_ref))
            .map(|line| {
                line.split_whitespace()
                    .next()
                    .expect("Git lines have sha then space")
            })
            .unwrap_or(&self.branch);
```

Reply:
Oh, yes this is much better!